### PR TITLE
Forward put_record requests to authorithy-discovery

### PR DESCRIPTION
**Pull request by alexggh**

### CODE REVIEW

File: substrate/client/authority-discovery/src/worker.rs [hunk 1]
```diff
@@ -581,7 +597,111 @@ where
 
 				debug!(target: LOG_TARGET, "Failed to put hash '{:?}' on Dht.", hash)
 			},
+			DhtEvent::PutRecordRequest(record_key, record_value, publisher, expires) => {
+				if let Err(e) = self
+					.handle_put_record_requested(record_key, record_value, publisher, expires)
+					.await
+				{
+					debug!(target: LOG_TARGET, "Failed to handle put record request: {}", e)
```
Comment (lexnv) [inline substrate/client/authority-discovery/src/worker.rs:605]: nit; this is unlikely to happen, should we bump this up to a warn for visibility? (similar to litep2p)

Comment (alexggh) [reply substrate/client/authority-discovery/src/worker.rs:605]: It is consistent with how we treat processing errors in this file, so I think debug is the right level here.

File: substrate/client/network/src/service.rs [hunk 2]
```diff
@@ -1311,6 +1325,7 @@ impl<'a> NotificationSenderReadyT for NotificationSenderReady<'a> {
 enum ServiceToWorkerMsg {
 	GetValue(KademliaKey),
 	PutValue(KademliaKey, Vec<u8>),
+	StoreValue(KademliaKey, Vec<u8>, Option<PeerId>, Option<Instant>),
```
Comment (skunert) [inline substrate/client/network/src/service.rs:1328]: This should be `StoreRecord`?

**Review by skunert — APPROVED** [review]
Looks good!

**Review by dmitry-markin — APPROVED** [review]

File: substrate/client/authority-discovery/src/worker.rs [hunk 3]
```diff
@@ -581,7 +597,111 @@ where
...
 		}
+
+		let authority_id =
+			self.known_authorities.get(&record_key).ok_or(Error::UnknownAuthority)?;
+		let signed_record =
+			Self::check_record_signed_with_authority_id(record_value.as_slice(), authority_id)?;
+		self.check_record_signed_with_network_key(
+			&signed_record.record,
+			signed_record.peer_signature,
+			publisher.ok_or(Error::MissingPublisher)?.into(),
```
[the remark this answers is not in the document]
Comment (alexggh) [reply substrate/client/authority-discovery/src/worker.rs]: Done.

File: substrate/client/network/src/service.rs (hunk 2 above)
Comment (alexggh) [reply substrate/client/network/src/service.rs]: Done.

**Pull request merged**